flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/delegation/ParserImplTest.java

```diff
@@ -0,0 +1,166 @@
+
+    private final boolean isStreamingMode = false;
+    private final TableConfig tableConfig = new TableConfig();
+    private final Catalog catalog = new GenericInMemoryCatalog("MockCatalog", "default");
+    private final CatalogManager catalogManager =
+            CatalogManagerMocks.preparedCatalogManager().defaultCatalog("builtin", catalog).build();
+    private final ModuleManager moduleManager = new ModuleManager();
+    private final FunctionCatalog functionCatalog =
+            new FunctionCatalog(tableConfig, catalogManager, moduleManager);
+    private final PlannerContext plannerContext =
+            new PlannerContext(
+                    tableConfig,
+                    functionCatalog,
+                    catalogManager,
+                    asRootSchema(new CatalogManagerCalciteSchema(catalogManager, isStreamingMode)),
+                    new ArrayList<>());
+
+    private final Supplier<FlinkPlannerImpl> plannerSupplier =
+            () ->
+                    plannerContext.createFlinkPlanner(
+                            catalogManager.getCurrentCatalog(),
+                            catalogManager.getCurrentDatabase());
+
+    private final Parser parser =
+            new ParserImpl(
+                    catalogManager,
+                    plannerSupplier,
+                    () -> plannerSupplier.get().parser(),
+                    t ->
+                            plannerContext.createSqlExprToRexConverter(
+                                    plannerContext.getTypeFactory().buildRelNodeRowType(t)));
+
+    private List<TestSpec> testLegalStatements;
+    private List<TestSpec> testIllegalStatements;
+
+    @Before
+    public void setup() {
+        testLegalStatements =
+                Arrays.asList(
+                        TestSpec.forStatement("ClEaR").expectedSummary("CLEAR"),
+                        TestSpec.forStatement("hElP").expectedSummary("HELP"),
+                        TestSpec.forStatement("qUIT").expectedSummary("QUIT"),
+                        TestSpec.forStatement("ExIT").expectedSummary("EXIT"),
+                        TestSpec.forStatement("REsEt").expectedSummary("RESET"),
+                        TestSpec.forStatement("   SEt ").expectedSummary("SET"),
+                        TestSpec.forStatement("SET execution.runtime-type=batch")
+                                .expectedSummary("SET execution.runtime-type=batch"),
+                        TestSpec.forStatement("SET pipeline.jars = /path/to/test-_-jar.jar")
+                                .expectedSummary("SET pipeline.jars=/path/to/test-_-jar.jar"),
+                        TestSpec.forStatement("USE test.db1").expectedSummary("USE test.db1"),
+                        TestSpec.forStatement("SHOW tables").expectedSummary("SHOW TABLES"));
+
+        testIllegalStatements =
+                Arrays.asList(
+                        TestSpec.forStatement("SET pipeline.name = ' '"),
```

Comment:
We should support this.

Reply:
If it's not easy to support this, we can create an issue to track this, and add a TODO comment above this.